src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1024,16 +1024,24 @@ private function registerPropertyInfoConfiguration(array $config, ContainerBuild
     private function registerCacheConfiguration(array $config, ContainerBuilder $container, XmlFileLoader $loader)
     {
         if (!empty($config['pools'])) {
-            $loader->load('cache_adapters.xml');
+            $loader->load('cache_pools.xml');
         }
 
         foreach ($config['pools'] as $name => $poolConfig) {
-            $poolDefinition = new DefinitionDecorator('cache.adapter.'.$poolConfig['type']);
-            $poolDefinition->replaceArgument(1, $poolConfig['default_lifetime']);
+            $type = $poolConfig['type'];
+            if (!isset($config['pools'][$type]) && !$container->has('cache.pool.'.$type)) {
+                throw new \UnexpectedValueException(sprintf('Cache pool "%s" is of undefined type "%s".', $name, $type));
+            }
+            $poolDefinition = new DefinitionDecorator('cache.pool.'.$type);
+            $poolDefinition->setPublic(!empty($poolConfig['public']));
```

Comment:
Is that possible? How can that happen if we don't do that in the extension class?